tests/gatewayapitests/npltests/gateway_npl_test.go

```diff
@@ -0,0 +1,453 @@
+	os.Setenv("POD_NAME", "ako-0")
+	os.Setenv("SERVICE_TYPE", "NodePortLocal")
+
+	// Set the user with prefix
+	_ = lib.AKOControlConfig()
+	lib.SetAKOUser(akogatewayapilib.Prefix)
+	lib.SetNamePrefix(akogatewayapilib.Prefix)
+	lib.AKOControlConfig().SetIsLeaderFlag(true)
+	akoControlConfig := akogatewayapilib.AKOControlConfig()
+	akoControlConfig.SetEventRecorder(lib.AKOGatewayEventComponent, tests.KubeClient, true)
+	registeredInformers := []string{
+		utils.ServiceInformer,
+		utils.SecretInformer,
+		utils.NSInformer,
+		utils.PodInformer,
+	}
+	utils.AviLog.SetLevel("DEBUG")
+	utils.NewInformers(utils.KubeClientIntf{ClientSet: tests.KubeClient}, registeredInformers, make(map[string]interface{}))
+	data := map[string][]byte{
+		"username": []byte("admin"),
+		"password": []byte("admin"),
+	}
+	object := metav1.ObjectMeta{Name: "avi-secret", Namespace: utils.GetAKONamespace()}
+	secret := &corev1.Secret{Data: data, ObjectMeta: object}
+	tests.KubeClient.CoreV1().Secrets(utils.GetAKONamespace()).Create(context.TODO(), secret, metav1.CreateOptions{})
+
+	akoApi := integrationtest.InitializeFakeAKOAPIServer()
+	defer akoApi.ShutDown()
+
+	tests.NewAviFakeClientInstance(tests.KubeClient)
+	defer integrationtest.AviFakeClientInstance.Close()
+
+	ctrl = akogatewayapik8s.SharedGatewayController()
+	ctrl.DisableSync = false
+	ctrl.InitGatewayAPIInformers(tests.GatewayClient)
+	akoControlConfig.SetGatewayAPIClientset(tests.GatewayClient)
+
+	stopCh := utils.SetupSignalHandler()
+	ctrlCh := make(chan struct{})
+	quickSyncCh := make(chan struct{})
+
+	waitGroupMap := make(map[string]*sync.WaitGroup)
+	wgIngestion := &sync.WaitGroup{}
+	waitGroupMap["ingestion"] = wgIngestion
+	wgFastRetry := &sync.WaitGroup{}
+	waitGroupMap["fastretry"] = wgFastRetry
+	wgSlowRetry := &sync.WaitGroup{}
+	waitGroupMap["slowretry"] = wgSlowRetry
+	wgGraph := &sync.WaitGroup{}
+	waitGroupMap["graph"] = wgGraph
+	wgStatus := &sync.WaitGroup{}
+	waitGroupMap["status"] = wgStatus
+
+	integrationtest.AddConfigMap(tests.KubeClient)
+	go ctrl.InitController(k8s.K8sinformers{Cs: tests.KubeClient}, registeredInformers, ctrlCh, stopCh, quickSyncCh, waitGroupMap)
+	os.Exit(m.Run())
+
+}
+
+const (
+	defaultPodName  = "test-pod"
+	defaultNodeName = "test-node"
+	defaultPodPort  = 80
+	defaultHostIP   = "10.10.10.10"
+	defaultPodIP    = "192.168.32.10"
+	defaultNodePort = 40001
+)
+
+func createPodWithNPLAnnotation(labels map[string]string) {
+	testPod := getTestPod(labels)
+	ann := make(map[string]string)
+	ann[lib.NPLPodAnnotation] = "[{\"podPort\":8080,\"nodeIP\":\"10.10.10.10\",\"nodePort\":60001}]"
+	testPod.Annotations = ann
+	tests.KubeClient.CoreV1().Pods(DEFAULT_NAMESPACE).Create(context.TODO(), &testPod, metav1.CreateOptions{})
+}
+
+func updatePodWithNPLAnnotation(labels map[string]string) {
+	testPod := getTestPod(labels)
+	ann := make(map[string]string)
+	ann[lib.NPLPodAnnotation] = "[{\"podPort\":8080,\"nodeIP\":\"10.10.10.10\",\"nodePort\":40001}]"
```

Comment:
Port value change mighe be required here as well

Reply:
changed to 60001